### PR TITLE
refactor(shell-proxy)!: rename env vars to `SHELLPROXY_*` and add usage message

### LINKED ISSUE #10455: shell-proxy plugin use DEFAULT_PROXY cause python urllib to use proxy anyway
### Describe the bug

I am using `shell-prxoy` plugin with [method 1](https://github.com/ohmyzsh/ohmyzsh/blob/bf303965e685489f2f1b764d5d22dc4766ca78c8/plugins/shell-proxy/README.md). setting `$DEFAULT_PROXY` to my proxy url.

But I found the name `DEFAULT_PROXY` will make python urllib to use proxy anyway.

### Steps to reproduce

1. Setting DEFAULT_PROXY
2. python urllib will use proxy anyway. for example `python -c 'import urllib.request; print(urllib.request.urlopen("file:///dev/urandom").read(2))'`

### Expected behavior

use proxy only if I enabled the proxy.

### Screenshots and recordings

_No response_

### OS / Distro

ArchLinux

### Zsh version

5.8

### Terminal emulator

deepin-terminal-gtk

### If using WSL on Windows, which version of WSL

_No response_

### Additional context

As the [document of python3 urllib.request](https://docs.python.org/3/library/urllib.request.html#urllib.request.getproxies) said. 
It has a `getproxies` method to scan environment variables [in a case insensitive way](https://github.com/python/cpython/blob/f4c03484da59049eb62a9bf7777b963e2267d187/Lib/urllib/request.py#L2508). with `DEFAULT_PROXY` it produce 
output like this.
```
{'default': '<http://my_proxy_url>'}
```
Then in https://github.com/python/cpython/blob/f4c03484da59049eb62a9bf7777b963e2267d187/Lib/urllib/request.py#L797-L804 it set a `'%s_open'` attribute in the object, this case `default_open` is override.

But this `default_open` method is being called before any `<protocol>_open()`.

image
Source: https://docs.python.org/3/library/urllib.request.html#urllib.request.OpenerDirector.open

This cause python urllib to use proxy anyway. When I setting a http proxy which doesn't support `file://` protocol it will fail.
image

Maybe we should use another variable name (no _proxy suffix) and show a deprecated log to `DEFAULT_PROXY` when using this plugin

### PR DESCRIPTION
**Pull request by fengkx**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.

## Changes:

- ~~use `__DEFAULT_PROXY__` instead of `DEFAULT_PROXY` to fix #10455~~
- Rename `DEFAULT_PROXY` to `SHELLPROXY_URL`
- Rename `CONFIG_PROXY` to `SHELLPROXY_CONFIG`
- Add usage prompt when no env vars are set
- Add deprecation message to old env variable names

Fixes #10455

### CODE REVIEW

File: plugins/shell-proxy/proxy.py [hunk 1]
```diff
@@ -5,11 +5,12 @@
 
 cwd = os.path.dirname(__file__)
 ssh_agent = os.path.join(cwd, "ssh-agent.py")
-user_proxy = os.environ.get("CONFIG_PROXY", os.path.expandvars("$HOME/.config/proxy"))
+user_proxy = os.environ.get(
+    "CONFIG_PROXY", os.path.expandvars("$HOME/.config/proxy"))
```
Comment (septs) [inline plugins/shell-proxy/proxy.py:9]: plz revert.

File: plugins/shell-proxy/proxy.py [hunk 2]
```diff
@@ -5,11 +5,12 @@
...
 
 
 def get_http_proxy():
-    default_proxy = os.environ.get("DEFAULT_PROXY")
+    default_proxy = os.environ.get("__DEFAULT_PROXY__")
```
Comment (septs) [inline plugins/shell-proxy/proxy.py:13]:
```suggestion
    default_proxy = os.environ.get("__DEFAULT_PROXY__") or os.environ.get("DEFAULT_PROXY")
```

File: plugins/shell-proxy/proxy.py [hunk 3]
```diff
@@ -18,7 +19,8 @@ def get_http_proxy():
 
 
 def make_proxies(url: str):
-    proxies = {"%s_PROXY" % _: url for _ in ("HTTP", "HTTPS", "FTP", "RSYNC", "ALL")}
+    proxies = {"%s_PROXY" % _: url for _ in (
+        "HTTP", "HTTPS", "FTP", "RSYNC", "ALL")}
```
Comment (septs) [inline plugins/shell-proxy/proxy.py:23]: plz revert.

File: plugins/shell-proxy/shell-proxy.plugin.zsh [hunk 4]
```diff
@@ -2,9 +2,15 @@
...
 proxy() {
-	source <(env "DEFAULT_PROXY=$DEFAULT_PROXY" "$__PROXY__" "$1")
+  local proxy_url
+  if [ -n "$DEFAULT_PROXY" ]; then 
+    echo "DEFAULT_PROXY is deprecated, use __DEFAULT_PROXY__ instead"
+    local proxy_url=$DEFAULT_PROXY
+  else
+    local proxy_url=$__DEFAULT_PROXY__
+  fi
+	source <(env "__DEFAULT_PROXY__=$proxy_url" "$__PROXY__" "$1")
```
Comment (septs) [inline plugins/shell-proxy/shell-proxy.plugin.zsh:13]:
```suggestion
  if [ -n "$DEFAULT_PROXY" ]; then
    echo "DEFAULT_PROXY is deprecated, use __DEFAULT_PROXY__ instead"
  fi
	source <(env "DEFAULT_PROXY=$DEFAULT_PROXY" "__DEFAULT_PROXY__=$__DEFAULT_PROXY__" "$__PROXY__" "$1")
```

File: plugins/shell-proxy/shell-proxy.plugin.zsh [hunk 5]
```diff
@@ -2,9 +2,15 @@
 # shellcheck disable=SC1090
 
 __PROXY__="${0:A:h}/proxy.py"
-
```
Comment (septs) [inline plugins/shell-proxy/shell-proxy.plugin.zsh:5]: plz revert the line.

**Comment by septs** [comment]
approved, plz wait for oh-my-zsh member approve

**Comment by mcornella** [comment]
Renamed `DEFAULT_PROXY` to `SHELLPROXY_URL` and `CONFIG_PROXY` to `SHELLPROXY_CONFIG`. I also added a usage message to `proxy.py` and refactored the code heavily. Test that it still works and this has my +1.

File: plugins/shell-proxy/README.md [hunk 6]
```diff
@@ -45,8 +59,4 @@ The working path of **Method 2** can be changed via `$CONFIG_PROXY`
 
 ## Maintainer
 
-- <https://github.com/septs>
-
-## The oh-my-zsh plugin (shell-proxy)
-
-Public Domain
```
Comment (septs) [inline plugins/shell-proxy/README.md:52]: plz keep the license statement

File: plugins/shell-proxy/proxy.py [hunk 7]
```diff
@@ -5,16 +5,22 @@
...
+proxy_config = os.environ.get("SHELLPROXY_CONFIG") or os.path.expandvars("$HOME/.config/proxy")
 
+usage="""shell-proxy: no proxy configuration found.
+
+Set `{env}` or create a config file at `{config}`
+See the plugin README for more information.""".format(env=proxy_env, config=proxy_config)
 
 def get_http_proxy():
-    default_proxy = os.environ.get("DEFAULT_PROXY")
+    default_proxy = os.environ.get(proxy_env)
```
Comment (septs) [inline plugins/shell-proxy/proxy.py:17]:
```suggestion
    default_proxy = os.environ.get("SHELLPROXY_URL")
```

File: plugins/shell-proxy/shell-proxy.plugin.zsh [hunk 8]
```diff
@@ -1,16 +1,37 @@
 #!/usr/bin/bash
```
Comment (septs) [inline plugins/shell-proxy/shell-proxy.plugin.zsh:1]: plz use [shellcheck](https://www.shellcheck.net) lint the shell script.
plz use [shfmt](https://github.com/mvdan/sh) format the shell script.

File: plugins/shell-proxy/README.md (hunk 6 above)
Comment (mcornella) [reply plugins/shell-proxy/README.md:52]: That's not a license statement. What do you mean by Public Domain?

File: plugins/shell-proxy/shell-proxy.plugin.zsh (hunk 8 above)
Comment (mcornella) [reply plugins/shell-proxy/shell-proxy.plugin.zsh:1]: `shellcheck` already passes, `shfmt` only does this:
```diff
diff --git a/plugins/shell-proxy/shell-proxy.plugin.zsh b/plugins/shell-proxy/shell-proxy.plugin.zsh
index bf0f4b0c..9d45b526 100644
--- a/plugins/shell-proxy/shell-proxy.plugin.zsh
+++ b/plugins/shell-proxy/shell-proxy.plugin.zsh
@@ -21,8 +21,8 @@ proxy() {
 
   # capture the output of the proxy script and bail out if it fails
   local output
-  output="$(SHELLPROXY_URL="$SHELLPROXY_URL" SHELLPROXY_CONFIG="$SHELLPROXY_CONFIG" "$proxy" "$1")" \
-  || return $?
+  output="$(SHELLPROXY_URL="$SHELLPROXY_URL" SHELLPROXY_CONFIG="$SHELLPROXY_CONFIG" "$proxy" "$1")" ||
+    return $?
 
   # evaluate the output generated by the proxy script
   source <(echo "$output")
```

**Comment by jsoref** [comment]
Hi, you're going to need to rebase this PR onto master in order to pass the spell checker which is going to ask you to add `SHELLPROXY` to `.github/actions/spelling/expect.txt`

**Comment by fengkx** [comment]
It looks like already rebased

**Comment by jsoref** [comment]
https://github.com/fengkx/ohmyzsh/tree/fix/shell-proxy-default-env
image

To make things happy you need to have 841f3cb0bb7663fa1062ffc59acb7b4581dc1d0f.

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork

**Comment by fengkx** [comment]
Forgot to fetch upstream, rebased

File: plugins/shell-proxy/README.md (hunk 6 above)
Comment (mcornella) [reply plugins/shell-proxy/README.md:52]: Hey @septs, should I just add this back? I'm asking whether there's a more clear licensing statement that you need here. What's the meaning of that to you? Is it like a `do whatever you want with this` kind of license?

Note that the project is licensed under the MIT license, which kinda already does that, just making sure that the copyright statement is kept. You can [browse these other licenses if you need](https://choosealicense.com/licenses/). Anyway, let me know.

File: plugins/shell-proxy/proxy.py (hunk 7 above)
Comment (mcornella) [reply plugins/shell-proxy/proxy.py:17]: Is this resolved?

File: plugins/shell-proxy/shell-proxy.plugin.zsh (hunk 8 above)
Comment (septs) [reply plugins/shell-proxy/shell-proxy.plugin.zsh:1]: plz use `shfmt` the script, thanks

File: plugins/shell-proxy/README.md (hunk 6 above)
Comment (septs) [reply plugins/shell-proxy/README.md:52]: ACK

**Review by mcornella — APPROVED** [review]

**Pull request merged**